distribution/sc-proxy-dex/src/proxy_farm.rs

```diff
@@ -160,6 +166,34 @@ pub trait ProxyFarmModule: proxy_common::ProxyCommonModule + proxy_pair::ProxyPa
         let farmed_token_returned = farm_result.0;
         let reward_token_returned = farm_result.1;
 
+        if reward_token_returned.token_id == farmed_token_returned.token_id {
```

Comment:
please refactor lines 169 to 189 - we might find something easier. put into a separate function and refactor the if else structures.

Reply:
fixed